packages/block-library/src/audio/edit.js

```diff
@@ -140,6 +174,19 @@ function AudioEdit( {
 
 	return (
 		<>
+			<BlockControls group="block">
+				<ToolbarButton
+					onClick={ () => {
+						setShowCaption( ! showCaption );
+						if ( showCaption && caption ) {
+							setAttributes( { caption: undefined } );
```

Comment:
I wasn't expecting hiding the caption to be destructive action. It can easily cause unexpected content loss for users.

Reply:
Hm.. this reminds me a bit a recent discussion(which you also commented) about whether the caption should be replaced or not when we replace an image(I couldn't find the link though..). I don't think any approach resonates with every user 🤷 . The change is kept in history and a user could undo. Furthermore for Audio and Video we didn't even add the caption if existed before, I added that here.. 

I don't have strong opinion, but this makes more sense to me.. Do you think adding a new attribute to control the display of the caption would be better?

Reply:
I believe it was #42834.

Maybe we can communicate better that the button adds or removes a caption and not just toggles the visibility.

@jasmussen, @jameskoster, what do you think?

Reply:
Tricky one. I assume the caption content is global, IE removing it in one post would remove it anywhere else the image is embedded across the site?

I think that would be fairly unexpected, and makes me lean towards the button being a visibility toggle rather than add/remove. 

> Maybe we can communicate better that the button adds or removes a caption and not just toggles the visibility.

Yes, at a minimum we should update the tooltip based on the button state.

Reply:
> Tricky one. I assume the caption content is global, IE removing it in one post would remove it anywhere else the media is embedded across the site?

No, captions are local to the posts.

Reply:
>No, captions are local to the posts.

Captions are local to each block. We could have the same audio with or without caption

Reply:
In that case it's probably fine to be an add/remove. Though if a caption exists in the Media Library, shouldn't that be added automatically when you click "Add caption"?

Reply:
>Though if a caption exists in the Media Library, shouldn't that be added automatically when you click "Add caption"?

If a caption exists is added when we first insert the media in the block automatically.

Reply:
Right, but if you then remove that caption (toggle off), then toggle it back on again, shouldn't it be populated by what is stored in the media library as a default?

Reply:
🤷 don't know what would be the best to be honest. I can implement this if you agree this is the desired behavior though 😄

Reply:
To be honest I think we're splitting hairs here. @Mamaduka is right to point out the the current behavior might be unexpected. But since all the other permutations can also be unexpected I don't think it's a big deal. Like you said before there is always to undo button.

Updating the button tooltip is probably all we need to do at this stage.

Reply:
Agreed. I updated the tooltip label to all three blocks that have this caption handling right now in trunk(audio, video, image).